backend/serve.go

```diff
@@ -31,5 +44,99 @@ func Serve(opts ServeOpts) error {
 		pluginOpts.TransformServer = newTransformSDKAdapter(opts.TransformDataHandler)
 	}
 
-	return grpcplugin.Serve(pluginOpts)
+	grpcplugin.Serve(pluginOpts)
+}
+
+type Plugin interface {
```

Comment:
Comments please for these exported interfaces and the types that Follow

Reply:
Yeah good point, but still a draft. So only consider the concept/examples